### ISSUE TITLE
Call hierarchy for variables, classes, macros, etc ...

### ISSUE THREAD

**Issue by talabir** [microsoft/vscode-cpptools#11037]
### Feature Request

In Eclipse CDT, one can ask for call hierarchy not only for functions but for variables, classes and much more.
The symbol appears as the root of the hierarchy, the references to that symbol appears in the next level of hierarchy and so on.

It is very useful feature for the common engineer, 
it could be great addition to the vscode features.

Thank you,
Tal.

**Comment by aniljosephk** [comment]
Yes. It's a much-awaited feature in vs code. I'm forced to use Eclipse CDT just for the call hierarchy feature.  Please implement this feature as soon as possible.

**Comment by sean-mcmanus** [comment]
@bobbrow Michelle and I discussed this -- it's just shows nothing for Calls To and shows the (confirmed) users of the "non-function code item" -- the advantage of Find All References is that the user functions can then be expanded to find their callers. So I don't think we need any more explanation from the users (I believe they're referring to the behavior for Java in Eclipse).

**Comment by talabir** [comment]
The ability to combine both find all references and call hierarchy is really useful.
For example, In case there are several references at the same function, the function appears only once and the number of references appear next to it

**Comment by sokurenko** [comment]
> The [Eclipse IDE documentation](https://help.eclipse.org/latest/index.jsp?topic=%2Forg.eclipse.cdt.doc.user%2Freference%2Fcdt_u_call_hierarchy_view.htm) only mentions functions for call hierarchy.
> 
> For variables, I'm not aware of there being a "hierarchy" but you can do "find all references" to see all the places it is used. Can you explain more how it works in Eclipse?
> 
> For classes, I see Eclipse has a "[Type Hierarchy](https://help.eclipse.org/latest/index.jsp?topic=%2Forg.eclipse.cdt.doc.user%2Freference%2Fcdt_u_type_hierarchy_view.htm&cp%3D9_4_0_13)" view that will let you see inheritance/interface implementation. That feature request is covered by #3799.

Please see https://github.com/microsoft/vscode-cpptools/issues/11765

**Comment by pushkarajthorat** [comment]
> The ability to combine both find all references and call hierarchy is really useful. For example, In case there are several references at the same function, the function appears only once and the number of references appear next to it

VScode should either warn saying, it is a variable and not a function hence Call Herarchie could not be shown. OR it should silently run Find All References whever variables are been searched upon for Call Herarchie.
